documentation/Using-the-SDK/Advanced/Adding-Column-Variants.md

```diff
@@ -0,0 +1,345 @@
+            {
+                return builder
+                    .WithToggle(
+                        new ColumnVariantDescriptor(new Guid("..."), "As Local Time"),
+                        column.Projector.Compose(t => t.ToLocalTime()));
+            });
+    }
+}
+```
+
+If there are cases where an added `DateTime` column needs additional variants outside of the ones applied in the extension method, you could define overloaded versions that take in an additional callback:
+
+```cs
+public static class TableBuilderExtensions
+{
+    public static void AddColumn(
+        this ITableBuilderWithRowCount tableBuilder,
+        ColumnConfiguration columnConfiguration,
+        IProjection<int, DateTime> dateTimeProjection)
+    {
+        tableBuilder.AddColumnWithVariants(columnConfiguration, dateTimeProjection, null);
+    }
+
+    public static void AddColumnWithVariants(
+        this ITableBuilderWithRowCount tableBuilder,
+        ColumnConfiguration columnConfiguration,
+        IProjection<int, DateTime> dateTimeProjection,
+        Func<ToggleableColumnBuilder, ColumnBuilder> columnBuilder)
+    {
+        tableBuilder.AddColumn(
+            new DataColumn<DateTime>(columnConfiguration, dateTimeProjection),
+            columnBuilder);
+    }
+
+    public static void AddColumn(
+        this ITableBuilderWithRowCount tableBuilder,
+        IDataColumn<DateTime> dateTimeColumn)
+    {
+        tableBuilder.AddColumn(dateTimeColumn, null);
+    }
+    
+    public static void AddColumnWithVariants(
+        this ITableBuilderWithRowCount tableBuilder,
+        IDataColumn<DateTime> dateTimeColumn,
+        Func<ToggleableColumnBuilder, ColumnBuilder> columnBuilder)
+    {
+        tableBuilder.AddColumnWithVariants(
+            dateTimeColumn,
+            builder =>
+            {
+                var withToggle = builder
+                    .WithToggle(
+                        new ColumnVariantDescriptor(new Guid("..."), "As Local Time"),
+                        dateTimeColumn.Projector.Compose(t => t.ToLocalTime()));
+
+                if (columnBuilder != null)
+                {
+                    return columnBuilder(withToggle);
+                }
+                else
+                {
+                    return withToggle;
+                }
+            });
+    }
+}
+```
+
+# Using Column Variants
+
+Registered column variants are exposed as `IDataColumn` instances where
+* The `IDataColumn.Configuration` is derived from the base column's configuration and the variant's `ColumnVariantDescriptor`
+* The data column's projection is the variant's projection
+
+Obtaining variant `IDataColumn` instances depends on how you are interacting with the SDK.
+
+## Using Column Variants via the Engine
+
+To use column variants in the SDK Engine, please refer to the "Using Column Variants" section of the [Using the Engine](../Using-the-engine.md#using-column-variants) documentation.
+
+## Using Column Variants via the SDK Runtime
```

Comment:
Is this something we want to document? I thought we only documented access through the Engine. Runtime shouldn't be documented because we allow breaking changes to the runtime.

Using the runtime directly is "at your own risk."

Reply:
Sure, I can remove it